retry/retry.go

```diff
@@ -0,0 +1,142 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+// Package retry provides a framework for retrying actions.
+// It does not itself invoke the action to be retried, but
+// is intended to be used in a retry loop.
+//
+// The basic usage is as follows:
+//
+//	for a := someStrategy.Start(); a.Next(); {
+//		try()
+//	}
+//
+// See examples for details of suggested usage.
+package retry
+
+import (
+	"time"
+
+	"github.com/juju/utils/clock"
+)
+
+// Strategy is implemented by types that represent a retry strategy.
+type Strategy interface {
+	NewTimer(now time.Time) Timer
```

Comment:
This type needs a much bigger explanation of how someone would implement a Strategy.  I have no idea what NewTimer is supposed to return.